aya/src/bpf.rs

```diff
@@ -580,44 +605,58 @@ impl<'a> BpfLoader<'a> {
                 (name, program)
             })
             .collect();
-        let maps: Result<HashMap<String, Map>, BpfError> = maps.drain().map(parse_map).collect();
+        let maps: Result<HashMap<String, Map>, BpfError> =
+            maps.drain().map(self.parse_map()).collect();
```

Comment:
Why not keep parse_maps as it is, and then just check for unsupported maps afterwards.

```suggestion
        let maps: Result<HashMap<String, Map>, BpfError> = maps.drain().map(parse_map).collect();
        if !self.allow_unsupported_maps && maps.iter().filter(|(_, x)| matches!(x, Map::Unsupported(_))).count() != 0 {
            return Err(BpfError::UnsupportedMap)
        }
```

Reply:
Done!